packages/common/core-interfaces/src/exposedUtilityTypes.ts

```diff
@@ -0,0 +1,549 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @rushstack/no-new-null */
+
+import type {
+	SerializationErrorPerNonPublicProperties,
+	SerializationErrorPerUndefinedArrayElement,
+} from "./jsonSerializationErrors.js";
+import type { JsonTypeWith, NonNullJsonObjectWith } from "./jsonType.js";
+
+/**
+ * Collection of utility types that are not intended to be used/imported
+ * directly outside of this package.
+ *
+ * @beta
+ * @system
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace InternalUtilityTypes {
+	/**
+	 * Returns non-symbol keys for optional properties of an object type.
+	 *
+	 * For homomorphic mapping use with `as` to filter. Example:
+	 * `[K in keyof T as NonSymbolWithOptionalPropertyOf<T, K>]: ...`
+	 *
+	 * @beta
+	 * @system
+	 */
+	export type NonSymbolWithOptionalPropertyOf<
+		T extends object,
+		Keys extends keyof T = keyof T,
+	> = Exclude<
+		{
+			[K in Keys]: T extends Record<K, T[K]> ? never : K;
+		}[Keys],
+		undefined | symbol
+	>;
+
+	/**
+	 * Returns non-symbol keys for required properties of an object type.
+	 *
+	 * For homomorphic mapping use with `as` to filter. Example:
+	 * `[K in keyof T as NonSymbolWithRequiredPropertyOf<T, K>]: ...`
+	 *
+	 * @beta
+	 * @system
+	 */
+	export type NonSymbolWithRequiredPropertyOf<
+		T extends object,
+		Keys extends keyof T = keyof T,
+	> = Exclude<
+		{
+			[K in Keys]: T extends Record<K, T[K]> ? K : never;
+		}[Keys],
+		undefined | symbol
+	>;
+
+	/**
+	 * Returns TTrue if T is sometimes serializable type, otherwise TFalse.
+	 * Fully not deserializable (functions, bigints, symbols, and undefined
+	 * less intersection with TException) produce TFalse.
+	 *
+	 * @beta
+	 * @system
+	 */
+	export type IfAtLeastSometimesDeserializable<T, TException, TTrue, TFalse> =
```

Comment:
Boy, naming these things is hard.  I think what this does is perform a "shallow" type test for serializability?  (e.g., an object or array are sometimes serializable depending on the types of their contents?)

Reply:
Definitely for complicated tests - which is most because otherwise we don't need a helper.
I will change this up a bit. I am realizing it is too easy to use incorrectly given its implementation. And I will update comments and perhaps the name.